src/python/grpcio_tests/tests_aio/unit/compression_test.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Tests behavior around the compression mechanism."""
+
+import asyncio
+import logging
+import platform
+import random
+import unittest
+
+import grpc
+from grpc.experimental import aio
+
+from tests_aio.unit._test_base import AioTestBase
+from tests_aio.unit import _common
+
+_GZIP_CHANNEL_ARGUMENT = ('grpc.default_compression_algorithm', 2)
+_GZIP_DISABLED_CHANNEL_ARGUMENT = ('grpc.compression_enabled_algorithms_bitset',
```

Comment:
Interesting way of testing things.

Reply:
I'm torn about doing the test in this way or the TCP proxy way. If we want to ensure the compression has worked 100% in unit test, I think the TCP proxy is the only way. So, this unit test is more or less a smoke test that tests if the compression arguments have been passed through.

BTW, we have compression interop test cases. However, since the compression bit is not exposed in Python layer, we didn't enable those tests.

If you feel strongly, I can write a proxy version of the compression unit tests.

Reply:
Up to you. It should take significantly less effort than the initial implementation, since it's already written, but what we really care about testing is that we have plumbed the right flags to core. We don't necessarily care about retesting that compression works; core should already be doing that in *its* tests.